### PR TITLE
Code cleanup and performance improvements

### PR DESCRIPTION
**Pull request by LiquidPT**
* Use the spack binary cache to download common packages recompiled
* Consolidate some installation code to common files
* Move paths added to the beginning for priority
* Upgrade GCC to a version commonly used in the spack binary cache, and enable a unit test to insure it's installed

### CODE REVIEW

**Review by abhamidipati-msft — APPROVED** [review]

File: tests/test-definitions.sh [hunk 1]
```diff
@@ -177,9 +177,9 @@ function verify_gcc_installation {
 # Check module file for the explicit installations
 function verify_gcc_modulefile {
     # Verify GCC Software installation path
-    check_exists "/opt/gcc-$gcc_version/"
+    check_exists "/opt/gcc-$gcc/"
```
Comment (yakovdyadkin) [inline tests/test-definitions.sh:180]: Should this still use `gcc_version`?

File: ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh [hunk 2]
```diff
@@ -23,7 +29,10 @@ $UBUNTU_COMMON_DIR/install_lustre_client.sh
 # install mellanox ofed
 ./install_mellanoxofed.sh
 
-# install nvidia gpu driver  
+# install mpi libraries
```
Comment (yakovdyadkin) [inline ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh:32]: Any particular reason for moving the MPI install earlier?

File: ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh
Comment (yakovdyadkin) [inline ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh:1]: Same comments as the ubuntu 20.04 `install.sh`

**Comment by yakovdyadkin** [comment]
General comment, can you squash/clean up commits?

File: common/install_spack.sh [hunk 3]
```diff
@@ -22,8 +22,11 @@ echo $source_spack_env | tee -a /etc/profile
 spack_version=$(spack --version | cut -d ' ' -f 1)
 $COMMON_DIR/write_component_version.sh "spack" $spack_version
 
+# Setup spack binary cache
```
[the remark this answers is not in the document]
Comment (LiquidPT) [reply common/install_spack.sh:25]: yes. they would download from the binary cache if available, and if not then use the existing functionality to download source and build

File: ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh (hunk 2 above)
Comment (LiquidPT) [reply ubuntu/ubuntu-20.x/ubuntu-20.04-hpc/install.sh:32]: Consistency with Alma mostly.

File: ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh
Comment (LiquidPT) [reply ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh:1]: see above comments

File: tests/test-definitions.sh (hunk 1 above)
Comment (LiquidPT) [reply tests/test-definitions.sh:180]: No. This is pulling from the component version file, and the keys are the component name.

**Pull request merged**